src/client/actions/ga/index.tsx

```diff
@@ -0,0 +1,45 @@
+import ReactGA from 'react-ga'
+import { get } from '../../util/requests'
+
+// Checks if reactga has initalised
+let hasInit = false
+
+const getGaId = async () => {
+    await get('/api/ga').then(async (response) => {
+      if(response.ok) {
+        await response.text().then((gaid) => {
+            ReactGA.initialize(gaid)
+            hasInit = true
+        })
+      }
+    })
+  }
+/**
+ * Send GA Event to record click - customise the category, action and label name to record multiple uniques per session
+ * @param {string} categoryName
+ * @param {string} eventName
+ * @param {string} [label='nothing']
+ */
+export const GAEvent = async (categoryName:string, eventName:string, label?: string| undefined) => {
+    if (!hasInit) {
+        await getGaId()
+    }
+    ReactGA.event({       
+        category: categoryName,  // Required
+        action: eventName,       // Required
+        label: label || 'nothing',       
```

Comment:
Actually how about lining up the param names against ReactGA, ie something like...
```suggestion
 * @param {string} category
 * @param {string} action
 * @param {string} [label='nothing']
 */
export const GAEvent = async (category: string, action: string, label?: string = 'nothing') => {
    if (!hasInit) {
        await getGaId()
    }
    ReactGA.event({       
        category,
        action,
        label,
```